onnxscript/tests/function_libs/torch_lib/ops_test.py

```diff
@@ -162,6 +162,11 @@ def run_test_output_match(
     # An example is nn.functional.upsample_nearest2d, which has a different signature
     # than the aten operator upsample_nearest2d
     onnx_function, input_wrangler = _split_function_and_wrangler(onnx_function_and_wrangler)
+    if not ops_test_common.dtype_op_schema_compatible(dtype, onnx_function.op_schema):
+        test_suite.skipTest(
+            f"dtype '{dtype}' is not supported by the op '{op.name}'. "
+            f"Type constraints: {onnx_function.op_schema.type_constraints}"
+        )
```

Comment:
I have a concern on skipping tests with the manual annotated type constraints. If we wrongly annotate an Op inputs, it could skip the tests it shouldn't, which makes us miss the very first opportunity to catch it.

Reply:
That's true. Any suggestions?

Reply:
wonder if we can xfail

Reply:
Based on what I found in https://github.com/pytorch/pytorch/pull/100265, indeed, if the type constraints are from onnx spec, most likely it's xfail-lable.

Reply:
Sometimes it will xpass if we do xfail because ort _can_ choose to implement the op for some types intentionally or unintentionally. So that's a little tricky

Reply:
How about put the logic after runs? Just a thought, if it fails, then we check the dtype is supported or not.

Reply:
Hmm but that’s still going to skip the tests where the annotation is incorrect?

Reply:
I thought about this again. Maybe that’s ok for now since if we declare that something is not supported via the annotations, then the exporter will just need to bridge the types via type promotion logic. When we see unnecessary type promotion in the graph we can then come back and make changes to the annotations.

Reply:
Merging for now. Feel free to share more thoughts!